src/input/agent/objective.rs

```diff
@@ -45,17 +47,35 @@ where
         // Check that required parameters are present and others are absent
         check_objective_parameter(&objective, &agent.decision_rule)?;
 
+        // Check that the year is a valid milestone year
+        ensure!(
+            milestone_years.contains(&objective.year),
```

Comment:
Because the years are sorted you can do a binary search:

```suggestion
            milestone_years.binary_search(&objective.year).is_ok(),
```

Bit of a micro-optimisation, but can't hurt.